src/main/java/org/quiltmc/loader/impl/discovery/ModResolver.java

```diff
@@ -120,143 +203,208 @@ private static IVecInt toVecInt(IntStream stream) {
 				if(!candidate.getInfo().getProvides().contains(s)) result.put(s, candidate);
 			}
 		} else {
-			// Inspired by http://0install.net/solver.html
-			// probably also horrendously slow, for now
+			Map<String, ModIdDefinition> modDefs = new HashMap<>();
+			DependencyHelper<LoadOption, ModLink> helper = new DependencyHelper<>(org.quiltmc.loader.util.sat4j.pb.SolverFactory.newLight());
```

Comment:
There a reason why this needs to be fully qualified?

Reply:
Sat4j has multiple classes called `SolverFactory`, both with a method called `newLight()`, however only the partial-boolean one works. (So using the full package name avoids accidentality using the wrong one). This is mostly just because I messed up once during development - however I'd be happy to change this to a normal import.